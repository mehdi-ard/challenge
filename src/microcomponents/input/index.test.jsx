import React from 'react';
import renderer from 'react-test-renderer';
import Input from '.';

it('renders default', () => {
  expect(
    renderer.create(<Input size="large" placeholder="large size" />).toJSON()
  ).toMatchSnapshot();
});
it('renders allowClear', () => {
  expect(
    renderer.create(<Input size="large" placeholder="large size" />).toJSON()
  ).toMatchSnapshot();
});

it('renders password', () => {
  expect(
    renderer
      .create(<Input type="password" size="large" placeholder="Password" />)
      .toJSON()
  ).toMatchSnapshot();
});

it('renders text area', () => {
  expect(
    renderer
      .create(<Input type="textarea" size="large" placeholder="Text Area" />)
      .toJSON()
  ).toMatchSnapshot();
});
it('renders search', () => {
  expect(
    renderer
      .create(<Input type="search" size="large" placeholder="search" loading />)
      .toJSON()
  ).toMatchSnapshot();
});
