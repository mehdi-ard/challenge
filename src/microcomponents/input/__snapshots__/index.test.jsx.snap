// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders allowClear 1`] = `
<input
  className="ant-input ant-input-lg"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="large size"
  type="text"
  value=""
/>
`;

exports[`renders default 1`] = `
<input
  className="ant-input ant-input-lg"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="large size"
  type="text"
  value=""
/>
`;

exports[`renders password 1`] = `
<span
  className="ant-input-password ant-input-password-large ant-input-affix-wrapper ant-input-affix-wrapper-lg"
>
  <input
    action="click"
    className="ant-input ant-input-lg"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Password"
    style={null}
    type="password"
    value=""
  />
  <span
    className="ant-input-suffix"
  >
    <i
      aria-label="icon: eye-invisible"
      className="anticon anticon-eye-invisible ant-input-password-icon"
      onClick={[Function]}
      onMouseDown={[Function]}
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="eye-invisible"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
        />
        <path
          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
        />
      </svg>
    </i>
  </span>
</span>
`;

exports[`renders search 1`] = `
<span
  className="ant-input-search ant-input-affix-wrapper ant-input-affix-wrapper-lg"
>
  <input
    className="ant-input ant-input-lg"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="search"
    style={null}
    type="text"
    value=""
  />
  <span
    className="ant-input-suffix"
  >
    <i
      aria-label="icon: loading"
      className="anticon anticon-loading ant-input-search-icon"
    >
      <svg
        aria-hidden="true"
        className="anticon-spin"
        data-icon="loading"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
      >
        <path
          d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
        />
      </svg>
    </i>
  </span>
</span>
`;

exports[`renders text area 1`] = `
<textarea
  className="ant-input"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Text Area"
  size="large"
  style={Object {}}
  value=""
/>
`;
